examples/bpi-monolith-test-suite/bpi/components/zkcircuits/circuitlib/orderCircuit.go

```diff
@@ -0,0 +1,182 @@
+package circuitlib
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"zkcircuit/json"
+
+	"github.com/consensys/gnark-crypto/ecc"
+	"github.com/consensys/gnark/backend"
+	"github.com/consensys/gnark/backend/groth16"
+	"github.com/consensys/gnark/frontend"
+	"github.com/consensys/gnark/std/hash/mimc"
+)
+
+type OrderCircuit struct {
+	I  Input
+	PI PrivateInput
+}
+
+type Input struct {
+	AgreementStateCommitment frontend.Variable `gnark:",public"`
+	StateObjectCommitment    frontend.Variable `gnark:",public"`
+	CalculatedAgreementRoot  frontend.Variable `gnark:",public"`
+}
+
+type PrivateInput struct {
+	O                  Order
```

Comment:
Just convention. Here a more golang-esque naming convention was used for those variables representing full type structs. It can be changed in a separate pull request for educational purposes as this will be the last PR for the monolith folder example that introduces functional changes.